### ISSUE TITLE
ts

### ISSUE THREAD

**Issue by iatsdotfatr** [Nirall/slider#29]
1. Добавить strict: true в tsconfig.
2. Не надо делать assertions, здесь просто явно приведи к строке.
![assertion](https://user-images.githubusercontent.com/51824674/110508207-f8efb280-812a-11eb-8dbd-999d3ef2b436.png)
3. По факту, тебе будет известен тип data в каждом конкретном случае применения. Замени any на дженерик.
![no any](https://user-images.githubusercontent.com/51824674/110508566-5421a500-812b-11eb-8fab-03f6b7a643cb.png)

**Comment by iatsdotfatr** [comment]
1. К строке лучше приводить через String.
2. Убрать any из проекта.
3. Почему undefined?
![undef](https://user-images.githubusercontent.com/51824674/111863751-95198500-8987-11eb-8759-2a7852115378.png)

**Comment by iatsdotfatr** [comment]
Почему тип method указан как string?
![obj string](https://user-images.githubusercontent.com/51824674/112946326-33130980-9157-11eb-82a5-96db658f6b66.png)

Если проблема с any в d.ts файле, попробуй его над его содержимым создать блок declare global {}.

**Comment by Nirall** [comment]
> Если проблема с any в d.ts файле, попробуй над его содержимым создать блок declare global {}.

Сделал по-другому, линтер вроде не ругается

**Comment by iatsdotfatr** [comment]
1. Зачем здесь дженерики? Просто объявляй тип данных как unknown и все.
![why generics](https://user-images.githubusercontent.com/51824674/115106864-08171b00-9f89-11eb-88d2-4bd03915a00d.png)

2. Здесь тоже зачем-то выводится intersection с T, хотя в функции выполняются другие действия. Тоже заменить на unknown.
![useless](https://user-images.githubusercontent.com/51824674/115106914-7956ce00-9f89-11eb-8b72-ad56742f4ca3.png)

3. Нет смысла делать union с unknown, так как любой тип совместим unknown, поэтому unknown по сути и есть union всех типов.
![Screenshot from 2021-04-17 14-31-28](https://user-images.githubusercontent.com/51824674/115107010-19acf280-9f8a-11eb-86d7-75f1091bd570.png)

4. Все входные данные должны быть нормализованы, то есть в этом метод, и ему подобные, уже должны передаваться валидные параметры, как и при инициализации.
![normalized](https://user-images.githubusercontent.com/51824674/115107056-5973da00-9f8a-11eb-9c8c-9e0f61d729c6.png)

**Comment by Nirall** [comment]
> 4\. Все входные данные должны быть нормализованы, то есть в этом метод, и ему подобные, уже должны передаваться валидные параметры, как и при инициализации.

Под нормализацией же понимаем только приведение данных к нужному типу, или нужна полноценная валидация с проверками типа minValue < maxValue?

**Comment by iatsdotfatr** [comment]
Да, только тип. И необязательно приведение, можно и ошибку выводить или выбрасывать.

**Comment by Nirall** [comment]
Подправил, для метода 'inputsAttach' ошибку выкидывать не стал, если не удовлетворяет условиям то просто не подключает инпуты.
